mkiso/mkiso.go

```diff
@@ -0,0 +1,159 @@
+package mkiso
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	diskfs "github.com/diskfs/go-diskfs"
+	diskpkg "github.com/diskfs/go-diskfs/disk"
+	"github.com/diskfs/go-diskfs/filesystem"
+	"github.com/diskfs/go-diskfs/filesystem/iso9660"
+	"github.com/diskfs/go-diskfs/partition/gpt"
+)
+
+func writeDiskFs(fs filesystem.FileSystem, file, diskPath string) error {
+
+	if path := filepath.Dir(diskPath); path != "/" {
+		if err := fs.Mkdir(path); err != nil {
+			return err
+		}
+	}
+
+	rw, err := fs.OpenFile(diskPath, os.O_CREATE|os.O_RDWR)
+	if err != nil {
+		return fmt.Errorf("failed to make %s on the disk image", diskPath)
+	}
+	content, err := os.ReadFile(file)
+	if err != nil {
+		return fmt.Errorf("failed to read %s", filepath.Base(file))
+	}
+	_, err = rw.Write(content)
+	if err != nil {
+		return fmt.Errorf("failed to write %s", filepath.Base(file))
+	}
+	return nil
+}
+
+func mkvfat(out, binary, config string) error {
+	var (
+		espSize          int64 = 100 * 1024 * 1024     // 100 MB
+		diskSize         int64 = espSize + 4*1024*1024 // 104 MB
+		blkSize          int64 = 512
+		partitionStart   int64 = 2048
+		partitionSectors int64 = espSize / blkSize
+		partitionEnd     int64 = partitionSectors - partitionStart + 1
+	)
```

Comment:
I forgot the part where we should do an estimation based on the input files instead of hardcoded MAGIC VALUES